### PR TITLE
feat: test APM Server with TLS and a Token configured

### LINKED ISSUE #13: Add agent tests for different apm-server config.
Add tests for using TLS and secret token.

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It adds the flags needed to configure the APM Server to use TLS and a Token.

## Why is it important?

This will test the security configuration on the APM Server and Agents, the flags are added to the `ALL` test so it applies to all agents.

## Related issues
Closes elastic/apm-integration-testing#13 elastic/apm-integration-testing#221

### CODE REVIEW

File: .ci/scripts/all.sh [hunk 1]
```diff
@@ -6,6 +6,21 @@ test -z "$srcdir" && srcdir=.
...
+  --no-apm-server-dashboards \
+  --no-apm-server-self-instrument \
+  --with-agent-rumjs \
+  --with-agent-dotnet \
+  --with-agent-go-net-http \
+  --with-agent-nodejs-express \
+  --with-agent-ruby-rails \
+  --with-agent-java-spring \
+  --with-agent-python-django \
+  --with-agent-python-flask \
```
Comment (mdelapenya) [inline .ci/scripts/all.sh]: Small refactor for follow-ups: could we define an array of agents and loop them adding a `--with-agent-$agent`?

Comment (kuisathaverat) [reply .ci/scripts/all.sh]: IMHO this would add an unnecessary complexity so add 8 values to a variable.

Comment (mdelapenya) [reply .ci/scripts/all.sh]: SGTM 👍

**Comment by kuisathaverat** [comment]
There are some test errors that we have to care about it

**Comment by kuisathaverat** [comment]
for .NET we have to wait for https://github.com/elastic/apm-agent-dotnet/pull/465

**Comment by kuisathaverat** [comment]
The only thing that still is not finished is the .Net agent support for ignoring certificates validation. https://github.com/elastic/apm-agent-dotnet/pull/465

**Pull request merged**